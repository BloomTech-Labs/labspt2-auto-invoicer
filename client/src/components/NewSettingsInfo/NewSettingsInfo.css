.settings_info {
  display: flex;
  flex-direction: column;
  width:100%;
  
}

.settings_info input {
  margin: 0px auto;
  margin-bottom: 10px;
}
.settings_info p {
  font-size: 1.4rem;
  text-align: center;
  margin-top: 5px;
}
.settings_info button {
  margin: 0px auto;
  border: 1px inset gray;
  border-radius: 3px;
  font-size: 2rem;
  padding: 1px 0;
  width: 80px;
  box-shadow: 3px 3px 3px lightgray;
}

.settings_info button:hover {
  border: 1px outset gray;
  background: lightgray;
  cursor: pointer;
}
