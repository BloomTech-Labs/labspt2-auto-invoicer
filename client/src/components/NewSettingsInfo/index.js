import React from "react";
// import css here
import "./NewSettingsInfo.css";
import Button from "../reusableComponents/Button"
const NewSettingsInfo = props => {
  const {
    email,
    oldPassword,
    newPassword,
    confirmPassword
  } = props.settingsInfo;
  const { changeHandler } = props;
  return (
    <fieldset className="settings_info">
      <input
        type="email"
        onChange={changeHandler}
        name="email"
        value={email}
        placeholder="Email Address"
      />
      <input
        type="password"
        onChange={changeHandler}
        name="oldPassword"
        value={oldPassword}
        placeholder="Old Password"
      />
      <input
        type="password"
        onChange={changeHandler}
        name="newPassword"
        value={newPassword}
        placeholder="New Password"
      />
      <input
        type="password"
        onChange={changeHandler}
        name="confirmPassword"
        value={confirmPassword}
        placeholder="Confirm Password"
      />
      <Button>save</Button>
      <p>Password must contain at least 8 charachters</p>
    </fieldset>
  );
};

export default NewSettingsInfo;
