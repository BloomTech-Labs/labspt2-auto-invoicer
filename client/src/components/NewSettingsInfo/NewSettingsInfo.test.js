import React from "react";
import { render } from "react-testing-library";

import NewSettingsInfo from "../NewSettingsInfo";

describe("<NewSettingsInfo />", () => {
  it("renders New Settings Info", () => {
    render(
      <NewSettingsInfo
        settingsInfo={{
          email: "testing@gmail.com",
          oldPassword: "testing1",
          newPassword: "testing2",
          confirmPassword: "testing2"
        }}
      />
    );
  });
});
