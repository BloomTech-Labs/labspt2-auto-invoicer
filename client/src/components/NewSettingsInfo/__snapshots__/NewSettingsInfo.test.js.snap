// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewSettingsInfo /> renders New Settings Info 1`] = `
<DocumentFragment>
  <fieldset
    class="settings_info"
  >
    <input
      name="email"
      placeholder="Email Address"
      type="email"
      value="testing@gmail.com"
    />
    <input
      name="oldPassword"
      placeholder="Old Password"
      type="password"
      value="testing1"
    />
    <input
      name="newPassword"
      placeholder="New Password"
      type="password"
      value="testing2"
    />
    <input
      name="confirmPassword"
      placeholder="Confirm Password"
      type="password"
      value="testing2"
    />
    <button
      class="button"
      type="submit"
    >
      save
    </button>
    <p>
      Password must contain at least 8 charachters
    </p>
  </fieldset>
</DocumentFragment>
`;
